tests/python_callback_test.py

```diff
@@ -1114,6 +1114,17 @@ def fun(x):
       result += fun(jnp.ones((500, 500), jnp.complex64))[1]
     jax.block_until_ready(result)  # doesn't deadlock
 
+  def test_pure_callback_fastpath(self):
+    # Regression test for https://github.com/jax-ml/jax/issues/31319
+    @jax.jit
+    def f(x):
+      return jax.pure_callback(jnp.sin, x, x)
+
+    x = jax.numpy.arange(5.0)
+    f(x).block_until_ready()
+    with jax.no_tracing(True):
```

Comment:
a better test would be to do: `jtu.count_pjit_cpp_cache_miss` and run f(x) twice inside the scope and check if the count is 1

Reply:
Thanks, done!